mlflow/entities/metric.py

```diff
@@ -10,6 +10,11 @@ class Metric(_MLflowObject):
     def __init__(self, key, value, timestamp, step):
         self._key = key
         self._value = value
+
+        if not isinstance(timestamp, int):
```

Comment:
I'd consider casting to int and printing some warning log once. It's much more intuitive to do `timestamp = time.time()`

Reply:
Agreed. But note, you need to use `timestamp = time.time() * 1000` here to make the unit correct.

Reply:
Do you think it would also be useful to specify a `float` type on the constructor as well as adding a docstring?

Reply:
> specify a float type on the constructor as well as adding a docstring?

@ntakouris Can you elaborate on this?